### PR TITLE
Build precompiled sdk and analyzer summary for dartdevc

### PR DESCRIPTION
**Pull request by jonahwilliams**
This is required to compile dart:ui sources without running in kernel mode.

### CODE REVIEW

File: web_sdk/libraries.dart [hunk 1]
```diff
@@ -0,0 +1,310 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (yjbanov) [inline web_sdk/libraries.dart]: This file has suspicious code style and copyright year. Has it been branched from somewhere?

File: web_sdk/make_combined_sdk.dart [hunk 2]
```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:dev_compiler/src/analyzer/command.dart' as command; // ignore: uri_does_not_exist
+
+// Script for creating flutter analyzer summmary and precompiled SDK.
```
Comment (yjbanov) [inline web_sdk/make_combined_sdk.dart]: nit: "Script for creating" => "Creates"
nit: "flutter" => "Flutter"

File: web_sdk/make_combined_sdk.dart [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  libraries.copySync(path.join(tempDartSdk.path, '_internal', 'sdk_library_metadata', 'lib', 'libraries.dart'));
+
+  // Prevent regular compilation from leaking into flutter
+  final File ddcSummary =
+      File(path.join(tempDartSdk.path, '_internal', 'ddc_sdk.sum'));
+  final File jsSdk =
+      File(path.join(tempDartSdk.parent.path, 'js', 'amd', 'dart_sdk.js'));
+  final File jsSdkMap =
+      File(path.join(tempDartSdk.parent.path, 'js', 'amd', 'dart_sdk.js.map'));
+  if (ddcSummary.existsSync()) {
```
Comment (yjbanov) [inline web_sdk/make_combined_sdk.dart]: Why do we need to check for their existence? Are these things not always there? Maybe instead _assert_ that they exist?

File: web_sdk/make_combined_sdk.dart [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    ddcSummary.deleteSync();
+  }
+  if (jsSdk.existsSync()) {
+    jsSdk.deleteSync();
+  }
+  if (jsSdkMap.existsSync()) {
+    jsSdkMap.deleteSync();
+  }
+
+  // Execute the analyzer summary generation.
```
Comment (yjbanov) [inline web_sdk/make_combined_sdk.dart]: This seems to generate more than an analyzer summary, e.g. `dart_sdk.js`.

File: web_sdk/pubspec.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
-name: web_sdk_tests
-author: Flutter Authors <flutter-dev@googlegroups.com>
+name: web_sdk_tests
```
Comment (yjbanov) [inline web_sdk/pubspec.yaml]: trailing whitespace

File: web_sdk/pubspec.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
-name: web_sdk_tests
-author: Flutter Authors <flutter-dev@googlegroups.com>
+name: web_sdk_tests	
+author: Flutter Authors <flutter-dev@googlegroups.com>	
 
 dev_dependencies:
-  analyzer: any
-  test: any
+  analyzer: any
```
Comment (yjbanov) [inline web_sdk/pubspec.yaml]: trailing whitespace

File: web_sdk/make_combined_sdk.dart [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    'dart:html_common',
+    'dart:svg',
+    'dart:web_audio',
+    'dart:web_gl',
+    'dart:web_sql',
+    'dart:ui',
+  ]);
+  final int result = (await command.compile(args)).exitCode;
+  if (result != 0) {
+    throw 'SDK generation failed';
```
Comment (yjbanov) [inline web_sdk/make_combined_sdk.dart]: `throw 'SDK generation failed with exit code $result'`

May also be useful to print the command and args.

File: web_sdk/BUILD.gn [hunk 8]
```diff
@@ -52,6 +51,30 @@ copy("web_ui_library") {
   ]
 }
 
+prebuilt_dart_action("generate_flutter_web_summary") {
```
Comment (yjbanov) [inline web_sdk/BUILD.gn]: Not sure about the name, does it only generate summary or does it also generate the JS SDK?

File: web_sdk/make_combined_sdk.dart (hunk 3 above)
Comment (jonahwilliams) [reply web_sdk/make_combined_sdk.dart]: I might be paranoid, but I was noticing the analyzer not overwriting the existing files...

**Pull request merged**